src/gt4py/next/program_processors/runners/dace_fieldview/transformations/strides.py

```diff
@@ -540,31 +545,49 @@ def _gt_map_strides_into_nested_sdfg(
         inner_desc.set_shape(inner_desc.shape, new_strides)
 
         # Now find the free symbols that the new strides need.
-        new_strides_symbols: list[str] = []
+        #  Note that usually `free_symbols` returns `set[str]`, but here, because
+        #  we fall back on SymPy, we get back symbols. We will keep them, because
+        #  then we can use them to extract the type form them, which we need later.
+        new_strides_symbols: list[dace.symbol] = []
         for new_stride_dim in new_strides:
             if dace.symbolic.issymbolic(new_stride_dim):
-                new_strides_symbols.append(str(new_stride_dim))
-            else:
                 new_strides_symbols.extend(sym for sym in new_stride_dim.free_symbols)
+            else:
```

Comment:
After some thinking probably not.
It could be that because some things are handled less strict, it is allowed (if we have multiple levels of nesting), but I will put an assert there.